hack/test.sh

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bash
+
+# 
+# Runs a test of the knative serving installation by deploying and then 
+# invoking an http echoing server.
+#
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+TERM="${TERM:-dumb}"
+
+main() {
+  local em=$(tput bold)$(tput setaf 2)
+  local me=$(tput sgr0)
+
+  # Drop some debug in the event even the above excessive wait does not work.
+  echo "${em}Testing...${me}"
+
+  echo "${em}  Creating echo server${me}"
+  n=0
+  until [ $n -ge 10 ]; do
+    cat <<EOF | kubectl apply -f - && break
+  apiVersion: serving.knative.dev/v1
+  kind: Service
+  metadata:
+    name: echo
+    namespace: func
+  spec:
+    template:
+      spec:
+        containers:
+          - image: docker.io/jmalloc/echo-server
+EOF
+    echo "Retrying..."
+    sleep 5
+  done
```

Comment:
As with the loop above, there is no error condition expressed if the loop executes 10 times unsuccessfully. I imagine the `kubectl wait...` below will ultimately time out, but perhaps we can fail fast here?

Reply:
@lance You are correct that a fall-through will yield an error on the next call.

However, I agree that an explicit error when exceeding the loop retries is correct, for both loop situations, and I will add that.

An oversight I would not have made if not in bash! :)

Reply:
This has been fixed.  Thanks @lance